node/src/main/scala/io/casperlabs/node/api/graphql/schema/globalstate/types/package.scala

```
@@ -417,15 +417,15 @@ package object types {
 
   lazy val StoredValue = ObjectType(
     "StoredValue",
-    fields[Unit, cltype.StoredValue](
+    fields[Unit, cltype.StoredValueInstance](
       Field(
         "value",
         StoredValueUnion,
         resolve = _.value match {
-          case cltype.StoredValue.Contract(value) => value
-          case cltype.StoredValue.Account(value)  => value
-          case cltype.StoredValue.CLValue(value) =>
-            cltype.CLValue.instantiate(value).right.get match {
+          case cltype.StoredValueInstance.Contract(value) => value
```

the `value` here is the same type as it used to? And `StoredValueInstance` is existing only to get better error messages? Can we get rid of it?

---

I was worried about the `.right.get` in the `resolve` function. It seemed to me like `resolve` should be a total function and it possibly throwing exceptions would be bad.

---

Could you raise an error somewhere in the `GraphQLSchemaBuilder` when it's `Left`?

---

That is the approach I am taking now using the `StoredValueInstance`. I don't think there is a way to raise an error from in a `resolve` function itself.

---

It's not adding any functionality. `StoredValue` is turned into `StoredValueInstance` before it reaches the `resolve`. I am trying out something else right now. One sec.

---

OK. I see it now. Nevermind.

It would be nice to clean up this `cltype` package a bit. There Scala ADT is living together with `CLValue/StoredValue` but it's not a blocker by no means.